python/cudf/cudf/core/column/timedelta.py

```diff
@@ -266,22 +266,24 @@ def default_na_value(self):
     def time_unit(self):
         return self._time_unit
 
-    def fillna(self, fill_value):
-        if cudf.utils.utils.isnat(fill_value):
-            return _fillna_natwise(self)
-        col = self
-        if is_scalar(fill_value):
-            if isinstance(fill_value, np.timedelta64):
-                dtype = determine_out_dtype(self.dtype, fill_value.dtype)
-                fill_value = fill_value.astype(dtype)
-                col = col.astype(dtype)
-            if not isinstance(fill_value, cudf.Scalar):
-                fill_value = cudf.Scalar(fill_value, dtype=dtype)
-        else:
-            fill_value = column.as_column(fill_value, nan_as_null=False)
+    def fillna(self, fill_value=None, method=None):
+        if fill_value is not None:
+            if cudf.utils.utils.isnat(fill_value):
+                return _fillna_natwise(self)
+            col = self
+            if is_scalar(fill_value):
+                if isinstance(fill_value, np.timedelta64):
+                    dtype = determine_out_dtype(self.dtype, fill_value.dtype)
+                    fill_value = fill_value.astype(dtype)
+                    col = col.astype(dtype)
+                if not isinstance(fill_value, cudf.Scalar):
+                    fill_value = cudf.Scalar(fill_value, dtype=dtype)
+            else:
+                fill_value = column.as_column(fill_value, nan_as_null=False)
 
-        result = libcudf.replace.replace_nulls(col, fill_value)
-        return result
+            return libcudf.replace.replace_nulls(col, fill_value)
```

Comment:
Should we call `super().fillna()` here as well?

Reply:
So I think `col` is not the same as `self` anymore at this line after time-unit coersion at L278, which makes `super` not usable.

Reply:
Ah, I guess we should just invoke `ColumnBase.fillna()` directly then:

```
ColumnBase.fillna(col,  fill_value)
```